server/test/serverTest.js

```diff
@@ -584,6 +584,165 @@ describe('Chatbot server', () => {
       )
     })
 
+    it('should start a new session if it has been >= 2 hours since last update of most recent open session, with no battery level sent', async () => {
+      const delayMs = 2 * 60 * 60 * 1000 + 7 * 60 * 1000 // 2h, + 7 mins to compensate for reminder/fallback updates
+      const timeNow = await db.getCurrentTime()
+      const timeNowMs = Date.parse(timeNow)
+
+      sinon
+        .stub(db, 'getCurrentTime')
+        .onFirstCall()
+        .returns(timeNow)
+        .onSecondCall()
+        .returns(timeNowMs + delayMs)
+
+      // eslint-disable-next-line prettier/prettier
+      await chai
+        .request(server)
+        .post('/flic_button_press')
+        .set('button-serial-number', unit1SerialNumber)
+        .send()
+
+      expect(imports.braveAlerter.startAlertSession).to.be.calledOnce
+
+      // eslint-disable-next-line prettier/prettier
+      await chai
+        .request(server)
+        .post('/flic_button_press')
+        .set('button-serial-number', unit1SerialNumber)
+        .send()
+
+      expect(imports.braveAlerter.startAlertSession).to.have.been.calledTwice
+
+      db.getCurrentTime.restore()
```

Comment:
So this is a tricky one. This line won't be executed if any of the `expect` tests fails, which will can cause some hard-to-debug test failures in other tests. Normally, I try to put my calls to `restore()` in the `afterEach` function so that I know that it will be executed no matter what. In this case, you probably don't want to change the existing `beforeEach` and `afterEach` functions. But what you could do is put your four new tests into a new `describe` that has its own `afterEach` to restore the stub. This new `describe` can be nested right here where you've got the `it` s, and that way it should still have the same `beforeEach` and `afterEach` as before, but then just also have your new one. Let me know if you have any questions!

Reply:
Done :)

Reply:
Could you please push this change :smile:

Reply:
Oops yes 😓🤣

Reply:
Thanks!